### ISSUE TITLE
-Wmissing-braces warning with clang on MacOS

### ISSUE THREAD

**Issue by guillaumechereau** [mity/md4c#125]
Can be easily fixed by replacing `{0}` with `{}` as suggested by clang:

```
[ 12%] Building C object src/CMakeFiles/md4c.dir/md4c.c.o
/Users/guillaume/src/md4c/src/md4c.c:1580:35: warning: suggest braces around initialization of subobject [-Wmissing-  braces]
    MD_UNICODE_FOLD_INFO a_fi = { 0 };
                                  ^
                                  {}
/Users/guillaume/src/md4c/src/md4c.c:1581:35: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    MD_UNICODE_FOLD_INFO b_fi = { 0 };
                                  ^
                                  {}
```

**Comment by mity** [comment]
Unfortunately, empty braces are not really a valid C, [not even in C99](https://stackoverflow.com/questions/17589533/is-an-empty-initializer-list-valid-c-code). Many compilers support as an extension because it has been added into C++, but not all of them.

**Comment by guillaumechereau** [comment]
I see... Actually the suggested fix by clang is to use `{ {0} }`.  Would that work?

**Comment by mity** [comment]
Oops, a race condition. No `{}` would work, even when nested. But it should already be fixed in 70d0ef7.

**Comment by guillaumechereau** [comment]
Thanks! Perfect.  I directly compile md4c source alongside my code, so it's nice that it compiles with -Wall by default.
